src/ansys/systemcoupling/core/adaptor/impl/injected_commands.py

```diff
@@ -31,49 +32,85 @@
+    ) -> None: ...
+
+    def upload_file(
+        self,
+        file_name: str,
+        remote_file_name: Optional[str] = None,
+        overwrite: bool = False,
+    ) -> None: ...
+
+
 def get_injected_cmd_map(
     category: str,
-    root_object: Container,
+    session: SessionProtocol,
     part_mgr: ParticipantManager,
     rpc,
 ) -> Dict[str, Callable]:
-    """Gets a dictionary that maps names to functions that implement the injected command.
+    """Get a dictionary mapping names to functions that implement injected commands
+    for the specified API category.
+
+    Whereas the set of commands that exists by default on the API represents a relatively
+    mechanical exposure of native System Coupling commands to PySystemCoupling, the
+    "injected commands" that are returned from here are either *additional* commands
+    that have no counterpart in System Coupling or *overrides* to existing commands
+    that provide modified or extended behavior.
 
-    The map returned pertains to the commands in the specified category.
     """
     ret = {}
 
     if category == "setup":
+        # ``get_injected_cmd_map`` needs to be called during initialisation, where
+        # the session API roots are not necessarily available yet. We therefore
+        # defer their access via the lambda.
+        get_setup_root_object = lambda: session.setup
         ret = {
             "get_setup_summary": lambda **kwargs: rpc.GetSetupSummary(**kwargs),
             "get_status_messages": lambda **kwargs: get_status_messages(
-                rpc, root_object, **kwargs
+                rpc, get_setup_root_object(), **kwargs
             ),
             "add_participant": lambda **kwargs: _wrap_add_participant(
-                root_object, part_mgr, **kwargs
+                session, part_mgr, **kwargs
             ),
         }
 
     if category == "solution":
+        get_solution_root_object = lambda: session.solution
+        get_setup_root_object = lambda: session.setup
         ret = {
-            "solve": lambda **kwargs: _wrap_solve(root_object, part_mgr, **kwargs),
+            "solve": lambda **kwargs: _wrap_solve(
+                get_solution_root_object(), part_mgr, **kwargs
+            ),
             "interrupt": lambda **kwargs: rpc.interrupt(**kwargs),
             "abort": lambda **kwargs: rpc.abort(**kwargs),
         }
 
     if category == "case":
+        get_case_root_object = lambda: session.case
         ret = {
             "clear_state": lambda **kwargs: _wrap_clear_state(
-                root_object, part_mgr, **kwargs
+                get_case_root_object(), part_mgr, **kwargs
             )
         }
 
     return ret
 
 
 def _wrap_add_participant(
-    root_object: Container, part_mgr: ParticipantManager, **kwargs
+    session: SessionProtocol, part_mgr: ParticipantManager, **kwargs
 ) -> str:
+    setup = session.setup
     if session := kwargs.get("participant_session", None):
```

Comment:
Minor nitpick as I don't think it affects the code but it might be a good idea for the sake of preventing confusion to rename this variable from session now that we pass another parameter named session of a different type.

Reply:
That's a good point - it does need fixing.

Reply:
This has been addressed.